File: mavsim_matlab/message_types/msg_path.m
% msg_path
%   - message type for path following
%
% mavsim_matlab 
%     - Beard & McLain, PUP, 2012
%     - Update history:  
%         3/12/2019 - RWB
classdef msg_path
   %--------------------------------
    properties
        type
        airspeed
        line_origin
        line_direction
        orbit_center
        orbit_radius
        orbit_direction
        flag_path_changed
    end
    %--------------------------------
    methods
        %------constructor-----------
        function self = msg_path()
            % flag can equal 'line' or 'orbit
            self.type = 'line';
            %self.type = 'orbit';
            % the airspeed that is commanded along the path
            self.airspeed = 25;
            % the origin of the line segement (r)
            self.line_origin = [0; 0; 0];
            % a unit vector that describes the direction of the line (q)
            self.line_direction = [1; 0; 0]; % unit vector
            % center of the orbit (c)
            self.orbit_center = [0; 0; 0];  
            % radius of the orbit (rho)
            self.orbit_radius = 50;
            % orbit direction: 'CW'==clockwise, 'CCW'==counter clockwise
            self.orbit_direction = 'CW';    
            % flag that indicates that the path has changed.  Used to
            % signal a redraw in the viewer
            self.flag_path_changed = 0;
        end
    end
end

File: mavsim_python/message_types/msg_path.py
"""
msg_path
    - messages type for input to path follower
    
part of mavsim_python
    - Beard & McLain, PUP, 2012
    - Last update:
        3/11/2019 - RWB
"""
import numpy as np


class MsgPath:
    '''
        Message class that defines a path
        'line' paths are defined by
            airspeed
            line_origin
            line_direction
        'orbit' paths are defined by
            orbit center
            orbit radius
            orbit direction
        plot_updated is for drawing purposes
    '''
    def __init__(self):
        # type='line' means straight line following, type='orbit' means orbit following
        self.type = 'line'
        #self.type = 'orbit'
        # desired airspeed along the path
        self.airspeed = float(25)
        # origin of the straight path line (r)
        self.line_origin = np.array([[0.0, 0.0, 0.0]]).T
        # direction of line -unit vector- (q)
        self.line_direction = np.array([[1.0, 0.0, 0.0]]).T
        # center of the orbit (c)
        self.orbit_center = np.array([[0.0, 0.0, 0.0]]).T
        # radius of the orbit (rho)
        self.orbit_radius = float(50)
        # orbit direction: 'CW'==clockwise, 'CCW'==counter clockwise
        self.orbit_direction = 'CW'
        # flag that indicates that path has been plotted
        self.plot_updated = bool(False)
